### PR TITLE
Controller primary input

### PR DESCRIPTION
**Pull request by kepler471**
Please test this, you will need a controller. 

I have removed any mouse input, but it is possible to have them simultaneously available.
This was just a quick addition, to be able to show what it's like with a controller.

Also, there are small errors with the player fliipping its facing direction, but that is because I havent gone to change everywhere where the mouse is hooked up to, just wanted to get a prototype out there fo you to try.

༼ つ ◕_◕ ༽つ

### CODE REVIEW

File: Moonshot/player/PlayerArsenal.gd [hunk 1]
```diff
@@ -6,7 +6,7 @@ class LaserBlaster:
 	var name = 'laser_blaster'
 	var bullet = load("res://Combat/bullet.tscn")
 	var fire_speed = 0.25
-	var f_mag = 800
+	var f_mag = 1600
```
Comment (Jamesbarford) [inline Moonshot/player/PlayerArsenal.gd:9]: ello

File: Moonshot/player/player.gd [hunk 2]
```diff
@@ -38,21 +38,28 @@ var safety = false
 var dead = false
 var priority_animations = ["stagger", "dodge"]
 
+var mdiff : Vector2
+var mouse : Vector2
+var msspd : float
```
Comment (Jamesbarford) [inline Moonshot/player/player.gd:43]: what is `msspd`?

File: Moonshot/player/player.gd [hunk 3]
```diff
@@ -38,21 +38,28 @@ var safety = false
...
+#var pos: Vector2
+var joys : Vector2
```
Comment (Jamesbarford) [inline Moonshot/player/player.gd:45]: What is `joys` ?

**Review by Jamesbarford — CHANGES_REQUESTED** [review]

File: Moonshot/player/player.gd (hunk 2 above)
Comment (kepler471) [reply Moonshot/player/player.gd:43]: unused variable

Comment (kepler471) [reply Moonshot/player/player.gd:43]: it was for a different mouse implementation, which i cut out for this PR, but didnt cut out all the bits

File: Moonshot/player/player.gd (hunk 3 above)
Comment (kepler471) [reply Moonshot/player/player.gd:45]: tracks joystick position.
But did you test this? i wasnt putting this up for a code quality check

File: Moonshot/player/PlayerArsenal.gd (hunk 1 above)
Comment (kepler471) [reply Moonshot/player/PlayerArsenal.gd:9]: Did it feel nicer to you?

**Comment by kepler471** [comment]
#318 adds this, with improvements, and properly hooks it up to everyting mouse was connected to. #318 allows either controller input, or mouse and keyboard.

**Pull request closed without merging**